cmd/hz/generator/template.go

```diff
@@ -36,26 +36,40 @@ type TemplateConfig struct {
 	Layouts []Template `yaml:"layouts"`
 }
 
+const (
+	Skip   = "skip"
+	Cover  = "cover"
+	Append = "append"
+)
+
 type Template struct {
-	Path           string    `yaml:"path"`            // The generated path and its filename, such as biz/handler/ping.go
-	Delims         [2]string `yaml:"delims"`          // Template Action Instruction Identifier, default: "{{}}"
-	Body           string    `yaml:"body"`            // Render template, currently only supports go template syntax
-	CustomTemplate bool      `yaml:"custom_template"` // Whether it is a custom template, This configuration does not work in layout templates, default: false
-	DivideMethod   bool      `yaml:"divide_method"`   // When CustomTemplate is equal to true, all methods are traversed and code is generated.
+	Default        bool           // Is the default package template
+	Path           string         `yaml:"path"`            // The generated path and its filename, such as biz/handler/ping.go
+	Delims         [2]string      `yaml:"delims"`          // Template Action Instruction Identifier, default: "{{}}"
+	Body           string         `yaml:"body"`            // Render template, currently only supports go template syntax
+	LoopMethod     bool           `yaml:"loop_method"`     // Loop generate files based on "method"
+	LoopService    bool           `yaml:"loop_service"`    // loop generate files based on "service"
+	UpdateBehavior UpdateBehavior `yaml:"update_behavior"` // Update command behavior; 0:unchanged, 1:regenerate, 2:append
+}
+
+type UpdateBehavior struct {
+	Behavior string `yaml:"behavior"` // Update command behavior; skip/cover/append
+	// the following variables are used for append update
+	AppendKey string   `yaml:"append_key"` // Append content based in key; for example: 'method'/'service'
+	InsertKey string   `yaml:"insert_key"` // Insert content by "insert_key"
+	AppendTpl string   `yaml:"append_tpl"` // Append content if UpdateBehavior is "append"
+	ImportTpl []string `yaml:"import_tpl"` // Import insert template
 }
 
 // TemplateGenerator contains information about the output template
 type TemplateGenerator struct {
-	OutputDir string
-	Config    *TemplateConfig
-	Excludes  []string
-	tpls      map[string]*template.Template
-	dirs      map[string]bool
-
-	// custom template files and whether divide method
-	customTpls    map[string]*template.Template
-	pathNameTpls  map[string]*template.Template
-	divideMethods map[string]bool
+	OutputDir    string
+	Config       *TemplateConfig
+	Excludes     []string
+	tpls         map[string]*template.Template // template name -> template
+	tplsInfo     map[string]*Template          // template name -> template info, for getting more template info
```

Comment:
What is the different between tpls/tplsInfo?
Maybe add more comment to explain？

Reply:
done~